### PR TITLE
fix: initialization of Prime calculator form + add comments

### PR DESCRIPTION
**Pull request by therealemjy**
## Changes

- fix Prime calculator form initialization
- add comments

I'd also recommend decoupling the Form index file a bit, using hooks for example. Finally, we'll need to add tests to cover all the other scenarios added when the form was wired up.

### CODE REVIEW

File: src/pages/PrimeCalculator/Form/index.tsx [hunk 1]
```diff
@@ -389,16 +370,19 @@ export const Form: React.FC = () => {
                     ? t('primeCalculator.stakedTokens.infos.label')
                     : undefined
                 }
+                // This value is incorrect: it should correspond to the amount of tokens taken in
+                // consideration when calculating the Prime rewards. It corresponds to the amount of
+                // tokens staked by the user, up to the maximum allowed
```
Comment (gleiser-oliveira) [inline src/pages/PrimeCalculator/Form/index.tsx]: This info conditionally appears if the user inputs more than `primeMaximumStakedXvsTokens`, which is the top cap of XVS staked taken into account, it is informed by the contract

File: src/pages/PrimeCalculator/Form/index.tsx [hunk 2]
```diff
@@ -416,6 +400,9 @@ export const Form: React.FC = () => {
             control={control}
             render={({ field }) => (
               <Field
+                // This value is incorrect: it should correspond to the amount of tokens taken in
+                // consideration when calculating the Prime rewards. It corresponds to the amount of
+                // tokens supplied by the user, up to the cap defined by Prime
```
Comment (gleiser-oliveira) [inline src/pages/PrimeCalculator/Form/index.tsx]: The value returned by primeEstimationData.supplyCapTokens should top to its cap according to the amount of XVS staked in the estimation

File: src/pages/PrimeCalculator/Form/index.tsx [hunk 3]
```diff
@@ -450,6 +437,9 @@ export const Form: React.FC = () => {
             control={control}
             render={({ field }) => (
               <Field
+                // This value is incorrect: it should correspond to the amount of tokens taken in
+                // consideration when calculating the Prime rewards. It corresponds to the amount of
+                // tokens borrowed by the user, up to the cap defined by Prime
```
Comment (gleiser-oliveira) [inline src/pages/PrimeCalculator/Form/index.tsx]: Same here

File: src/pages/PrimeCalculator/Form/index.tsx [hunk 4]
```diff
@@ -188,76 +200,50 @@ export const Form: React.FC = () => {
...
+        xvs &&
+        getXvsVaultUserInfoData &&
+        userStakedXvsTokens &&
+        userStakedXvsMantissa?.isGreaterThanOrEqualTo(primeMinimumStakedXvsMantissa) &&
+        (selectedAsset?.userBorrowBalanceTokens.isGreaterThan(0) ||
+          selectedAsset?.userSupplyBalanceTokens.isGreaterThan(0))
+      ) {
+        setValue('stakedAmountXvsTokens', userStakedXvsTokens.toFixed());
+        setValue('suppliedAmountTokens', selectedAsset.userSupplyBalanceTokens.toFixed());
+        setValue('borrowedAmountTokens', selectedAsset.userBorrowBalanceTokens.toFixed());
```
Comment (gleiser-oliveira) [inline src/pages/PrimeCalculator/Form/index.tsx]: It was requested to not reset `stakedAmountXvsTokens` when changing tokens, hence the

```
if (formState.touchedFields.stakedAmountXvsTokens)
  setValue('borrowedAmountTokens', borrowedAmountTokens);
  setValue('suppliedAmountTokens', suppliedAmountTokens);
```

Comment (therealemjy) [reply src/pages/PrimeCalculator/Form/index.tsx]: Yeah, to be honest that makes sense since the amount staked does not change.

**Comment by therealemjy** [comment]
Nice job, this looks very good 👍
I can't approve since I opened the PR, but feel free to approve and merge.

**Review by gleiser-oliveira — APPROVED** [review]

**Pull request merged**